### ISSUE TITLE
defineModel not working with eslint:recommended rules

### ISSUE THREAD

**Issue by yurks** [vuejs/eslint-plugin-vue#2392]
**Checklist**

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.21.1
- **Vue version:** 3.4.15
- **Node version:** 20
- **Operating System:** osx

**Please show your full configuration:**

```json5
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,

  extends: [
    'plugin:json/recommended',
    'plugin:vue/vue3-recommended',
    'plugin:jest-dom/recommended',
    'plugin:testing-library/vue',
    'eslint:recommended',
    '@vue/eslint-config-typescript/recommended',
    '@vue/eslint-config-prettier'
  ],
  rules: {
    'no-console': ['error', { allow: ['warn', 'error'] }],
    eqeqeq: ['error', 'always', { null: 'ignore' }],
    'vue/multi-word-component-names': 0,
    'vue/no-expose-after-await': 0,
    'vue/no-empty-component-block': 1
  },
  ignorePatterns: ['*.gen.ts']
}

```

**What did you do?**	

```vue
const model = defineModel<boolean>()
```

**What did you expect to happen?**
pass defineModel validation

**What actually happened?**

<img width="414" alt="image" src="https://github.com/vuejs/eslint-plugin-vue/assets/4581398/fc9d6ba0-20c1-4c6f-bfbf-2537c0f5b2a0">

**Repository to reproduce this issue**

**Comment by FloEdelmann** [comment]
`no-undef` is not an eslint-plugin-vue rule, but a core rule: https://eslint.org/docs/latest/rules/no-undef

You have to tell ESLint that `defineModel` is a global variable.

**Comment by yurks** [comment]
@FloEdelmann `defineProps` and `defineEmits` are global vars as well as new `defineModel`. But validation is passed for both of them without any additional setup.

**Comment by yurks** [comment]
the reason or that could be here https://github.com/vuejs/eslint-plugin-vue/blob/d1dd90e5c6d1c4115b9944fba89d952b8b93297b/lib/index.js#L268
`defineModel` is absent. 

Just wondering about deprecation notice here. In case that setup going to be deprecated, then I expect another `define*` will throw an error 😕

**Comment by FloEdelmann** [comment]
See https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-generate-no-undef-warnings; `vue/setup-compiler-macros` (the code you are referencing is belonging to that rule) is no longer necessary and thus deprecated. Upgrading eslint-plugin-vue (and transitively vue-eslint-parser) should do the trick.
